src/_guides/language/sound-problems.md

```diff
@@ -240,8 +240,8 @@ For more information, see [Use proper input parameter types when overriding meth
 **Warning:**
 <code>Unsound implicit cast from <em>Class&lt;dynamic&gt;</em> to <em>Class&lt;type&gt;</em>.</code>
```

Comment:
I think only the "Unsound implicit downcast" bit has changed.  It's not a static error or warning.

That said, most of your examples are still relevant if the downcast fails at runtime.